### PR TITLE
Update release instructions

### PR DESCRIPTION
**Pull request by adrianludwin**
Move the release instructions from the README to their own doc. Expand
with the latest on branching and doc updates.

/assign @yiqigao217 
cc @adrianchung @rjbez17 @GinnyJI

### CODE REVIEW

File: incubator/hnc/docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+# Releasing HNC
+
+HNC is uses semantic versioning. We use long-lived branches for every minor
```
Comment (yiqigao217) [inline incubator/hnc/docs/releasing.md:3]: remove 'is'.

File: incubator/hnc/docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+**Note that your personal access token will be visible in the build logs,** but
+will not be printed to the console from `make` itself. TODO: [fix
+this](https://cloud.google.com/cloud-build/docs/securing-builds/use-encrypted-secrets-credentials#example_build_request_using_an_encrypted_variable).
+
+## Test
+
+Test! At a minimum, install it onto a cluster, download the kubectl plugin, and
+ensure that you can create a subnamespace. Better yet, run through the demo and
+all the `hack/test-*.sh` scripts. _TODO: automate this_
```
Comment (yiqigao217) [inline incubator/hnc/docs/releasing.md:116]: Maybe add a link to the e2e test issue #785 ?

File: incubator/hnc/docs/releasing.md (hunk 1 above)
Comment (adrianludwin) [reply incubator/hnc/docs/releasing.md]: Embarrassing!

File: incubator/hnc/docs/releasing.md (hunk 2 above)
Comment (adrianludwin) [reply incubator/hnc/docs/releasing.md]: Done

**Pull request merged**